lib/src/main/scala/spinal/lib/bus/amba4/axi/Axi4SharedOnChipRam.scala

```diff
@@ -60,60 +60,123 @@ case class Axi4SharedOnChipRam(dataWidth : Int, byteCount : BigInt, idWidth : In
 }
 
 
-//WARNING, do not support backpresure
-case class Axi4SharedOnChipRamMultiPort(portCount : Int, dataWidth : Int,byteCount : BigInt,idWidth : Int) extends Component{
-  val axiConfig = Axi4SharedOnChipRam.getAxiConfig(dataWidth,byteCount,idWidth)
+case class Axi4SharedOnChipRamPort(config: Axi4Config) extends Bundle {
+    def attach(ram: Mem[Bits]): Axi4Shared = {
```

Comment:
One issue is that this function will generate much logic without naming any of its content.

What's about 
```scala
def attach(ram: Mem[Bits]): Axi4Shared = {
   val axi = Axi4Shared(config)
   val logic = new Composite(axi, "logic"){
      
      ...
   }
   axi
}
```

Reply:
OK, I am not very familiar with the name policy. However, attach a name is really great, while debugging.

Reply:
I have tried this Composite in attach. However, I could not find any difference after adding that.

Reply:
I suppose that it is because the usage of Axi4SharedOnChipRamPort, where a vector but a standalone signal is assigned.
And I also tried to create new area, unfortunately it does not helps.
```
val ports = Vec(Axi4SharedOnChipRamPort(config), portCount)
```

Reply:
I have tried if an individual assignment is applies as below, the name system works.
```
       val axis1 = ports(1).attach(ram)
```
But it would fail when
```
for (i <- 0 until portCount) {
        val axi = ports(i).attach(ram)
        io.axis(i) >> axi
}
```
So I suppose that the problem is the compatibility between name system and container iteration.

Reply:
very important to attache names ^^ 

```scala
val miaou = for (i <- 0 until portCount) yield new Area{
        val axi = ports(i).attach(ram)
        io.axis(i) >> axi
}
````

Reply:
The lazy initialization Handle strategy has been adopted to create the port logic while attaching. And it's still accessable through object property so that the name can be kept.
It seems that calling a function will not let the reference hold the name of logic.